File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/wowGameData/types/mythicKeystone/MythicKeystoneMemberModel.ts
import type { CharacterSpecializationModel } from '../character/CharacterSpecializationModel';
import type { WoWCharacterRaceModel } from '../character/WoWCharacterRaceModel';
import type { DungeonCharacterModel } from './DungeonCharacterModel';

export type MythicKeystoneMemberModel = {
    character: DungeonCharacterModel;
    specialization: CharacterSpecializationModel;
    race: WoWCharacterRaceModel;
    equippedItemLevel: number;
}

File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/WoWGameData/Character/MythicKeystone/MythicKeystoneMemberModel.cs
using System.Text.Json.Serialization;

namespace CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.MythicKeystone;

public class MythicKeystoneMemberModel
{
    [JsonPropertyName("character")]
    public DungeonCharacterModel Character { get; set; }

    [JsonPropertyName("specialization")]
    public CharacterSpecializationModel Specialization { get; set; }

    [JsonPropertyName("race")]
    public CharacterRaceModel Race { get; set; }

    [JsonPropertyName("equipped_item_level")]
    public int EquippedItemLevel { get; set; }
}
